File: eBarberShop/UI/ebarbershop_mobile/lib/models/rezervacija/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int rezervacijaId;
  DateTime datum;
  DateTime vrijeme;
  bool? status;
  int korisnikId;
  int uposlenikId;
  int uslugaId;

  Rezervacija(this.rezervacijaId, this.datum, this.vrijeme, this.status,
      this.korisnikId, this.uposlenikId, this.uslugaId);

  factory Rezervacija.fromJson(Map<String, dynamic> json) => _$RezervacijaFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}

File: eBarberShop/eBarberShop.Services/Database/Rezervacija.cs
namespace eBarberShop.Services.Database
{
    public partial class Rezervacija
    {
        public int RezervacijaId { get; set; }
        public DateTime Datum { get; set; }
        public DateTime Vrijeme { get; set; }
        public bool? Status { get; set; }
        public int KorisnikId { get; set; }
        public int UposlenikId { get; set; }
        public int UslugaId { get; set; }

        public virtual Korisnici Korisnik { get; set; }
        public virtual Uposlenik Uposlenik { get; set; }
        public virtual Usluga Usluga { get; set; }
    }
}
